packages/sitecore-jss-react/src/enhancers/usePersonalization.tsx

```diff
@@ -0,0 +1,54 @@
+import { LayoutPersonalizationService } from '@sitecore-jss/sitecore-jss';
+import { useEffect, createElement, useReducer } from 'react';
+import { MissingComponent } from '../components/MissingComponent';
+import { useComponentFactory } from './withComponentFactory';
+
+export function usePersonalization(props: { uid: string, layoutPersonalizationService: LayoutPersonalizationService, missingComponentComponent?: React.ComponentClass<unknown> | React.FC<unknown> }) {
+  // forceUpdate emulating, we need to re-render the component after personalization loading
+  const [, forceUpdate] = useReducer(x => x + 1, 0);
+  const componentFactory = useComponentFactory();
+
+  const personalizedComponentLayout = props.layoutPersonalizationService.getPersonalizedComponent(props.uid);
+  const isLoading = props.layoutPersonalizationService.isLoading();
+
+  useEffect(() => {
+    if (!isLoading) {
+      return;
+    }
```

Comment:
Add empty lines after closing brackets (not only here). Code, looking like a single piece, is difficult to read.

Reply:
done